pkgs/build-support/libredirect/default.nix

```diff
@@ -5,15 +5,14 @@ stdenv.mkDerivation {
 
   unpackPhase = "cp ${./libredirect.c} libredirect.c";
 
-  buildPhase =
-    ''
-      gcc -Wall -std=c99 -O3 -shared libredirect.c -o libredirect.so -fPIC -ldl
-    '';
+  buildPhase = ''
+    cc -Wall -std=c99 -O3 -shared libredirect.c -o libredirect.so -fPIC -ldl
+  '';
 
   installPhase = "mkdir -p $out/lib; cp libredirect.so $out/lib";
 
   meta = {
-    platforms = stdenv.lib.platforms.linux;
+    platforms = stdenv.lib.platforms.linux ++ stdenv.lib.platforms.darwin;
```

Comment:
I think in this case platforms.unix can be used. Usually if something supports Linux & Darwin, it will support any Unix system.

Reply:
Not all Unix systems have `LD_PRELOAD` or `DYLD_INSERT_LIBRARIES`, only those that use ELF (and well, Mach-O).